plugins/webp-uploads/tests/data/class-testcase.php

```diff
@@ -137,6 +137,20 @@ public function set_image_output_type( string $format ): void {
 	 * Opt into picture element output.
 	 */
 	public function opt_in_to_picture_element(): void {
-		add_filter( 'wp_content_img_tag', 'webp_uploads_wrap_image_in_picture', 10, 3 );
+		remove_filter( 'wp_content_img_tag', 'webp_uploads_wrap_image_in_picture', 10 );
+		remove_filter( 'the_content', 'webp_uploads_update_image_references', 13 ); // Run after wp_filter_content_tags.
+
+		// Apply picture element support.
+		update_option( 'webp_uploads_use_picture_element', '1' );
```

Comment:
Is enabling the `perflab_generate_webp_and_jpeg` option not needed?

```suggestion
		update_option( 'perflab_generate_webp_and_jpeg', '1' );
		update_option( 'webp_uploads_use_picture_element', '1' );
```

The `webp_uploads_is_picture_element_enabled()` function only returns true if both of these are enabled:

https://github.com/WordPress/performance/blob/e7313851f39749886a4ae0bacbcd620a6effce3d/plugins/webp-uploads/helper.php#L395-L408

Reply:
@westonruter https://github.com/WordPress/performance/blob/fix/1316-check-img-src/plugins/webp-uploads/tests/test-picture-element.php#L43-L44 we set fallback for all the tests so it generate JPEG also for all tests. Also in `test_maybe_wrap_images_in_picture_element` test we needs to update that option to no JPEG fallback so update option to return false.

Reply:
OK, got it. Still, it seems it should be added here for completeness.

Also, I believe that `wpSetUpBeforeClass` isn't the right place to set the option. It should be set in `set_up`. The reason for this is that `wpTearDownAfterClass` does not reset the database. So setting an option needs to be done in `set_up` instead, otherwise the option will leak outside the test. 

Secondly, I believe that `wpSetUpBeforeClass` should be calling `parent::wpSetUpBeforeClass()` as otherwise there is base class functionality which won't be getting called.

Thirdly, the current implementation of `wpSetUpBeforeClass` which is generating an image should have a corresponding `wpTearDownAfterClass` method (which also calls `parent::wpTearDownAfterClass()`) while also deleting that uploaded image. Otherwise the image will never get cleaned up, resulting in more and more wasted space with each test run.

Reply:
Correction: the base class doesn't actually define `wpSetUpBeforeClass()` or  `wpTearDownAfterClass` so no need to call them. See https://github.com/WordPress/performance/pull/1408#discussion_r1708658029

Reply:
> Also, I believe that `wpSetUpBeforeClass` isn't the right place to set the option. It should be set in `set_up`. The reason for this is that `wpTearDownAfterClass` does not reset the database. So setting an option needs to be done in `set_up` instead, otherwise the option will leak outside the test.

When i move the update option in `set_up` the JPEG fallback image is not returning. 🤔 

> Thirdly, the current implementation of `wpSetUpBeforeClass` which is generating an image should have a corresponding `wpTearDownAfterClass` method ~(which also calls `parent::wpTearDownAfterClass()`)~ while also deleting that uploaded image. Otherwise the image will never get cleaned up, resulting in more and more wasted space with each test run.

Done in https://github.com/WordPress/performance/pull/1408/commits/c693e96dd73bc75cdb68ece5cea711e9d99804c5

Reply:
> When i move the update option in `set_up` the JPEG fallback image is not returning. 🤔

Oh, right, because `set_up` happens after `wpSetUpBeforeClass`. So if you intend to set the option in `wpSetUpBeforeClass()` then you should be sure to delete it in `wpTearDownAfterClass()`.

Reply:
See https://github.com/WordPress/performance/pull/1408/files#r1708690705